proof/invariant-abstract/DetSchedDomainTime_AI.thy

```diff
@@ -18,193 +18,192 @@ definition
   "valid_domain_list_2 dlist \<equiv> 0 < length dlist \<and> (\<forall>(d,time) \<in> set dlist. 0 < time)"
 
 abbreviation
-  valid_domain_list :: "det_ext state \<Rightarrow> bool"
+  valid_domain_list :: "'z state \<Rightarrow> bool"
 where
-  "valid_domain_list \<equiv> (\<lambda>s. valid_domain_list_2 (domain_list s))"
+  "valid_domain_list \<equiv> \<lambda>s. valid_domain_list_2 (domain_list s)"
 
 lemmas valid_domain_list_def = valid_domain_list_2_def
 
 
 section \<open>Preservation of domain list validity\<close>
 
-lemma ethread_get_wp[wp]:
-  "\<lbrace>\<lambda>s. etcb_at (\<lambda>t. P (f t) s) ptr s\<rbrace> ethread_get f ptr \<lbrace>P\<rbrace>"
-  unfolding ethread_get_def
-  by (wp | clarsimp simp add: get_etcb_def etcb_at'_def is_etcb_at'_def)+
-
-(* We want wp to use ethread_get_inv before ethread_get_wp *)
-declare ethread_get_inv[wp del, wp]
-
+(*
+  FIXME: cleanup
+  Many of these could be factored out into the general state_ext class instead, but they will be
+  applied to det_ext lemmas that contain e.g. preemption_point which needs the det_ext work_units,
+  i.e. those will need additional locales, because 'state_ext needs to be interpreted first
+  into ?'state_ext.
+*)
```

Comment:
I'm not actually convinced that this factoring out would be worth it. We are in a `DetSched*` theory after all, and it would be perfectly fine for everything in here to be in `det`, even if some facts would also be true about `state_ext`.

Reply:
Right, but with this change the `DetSched*` theories don't actually depend on anything from the extended state, which is a bit unfortunate with the naming of them. I do agree anyway that this might not be worth it, although I would be more tempted by similar issues in locales in earlier files. They generally don't break the proofs but mean that lemmas need to be tagged with `det_state` a fair bit more than necessary.

Reply:
Right, that makes complete sense. Maybe filing an issue for that is the best idea then.